django/contrib/admin/static/admin/js/SelectBox.js

```diff
@@ -79,30 +79,26 @@
             return false;
         },
         move: function(from, to) {
-            var from_box = document.getElementById(from);
-            var option;
-            var boxOptions = from_box.options;
-            for (var i = 0, j = boxOptions.length; i < j; i++) {
-                option = boxOptions[i];
-                if (option.selected && SelectBox.cache_contains(from, option.value)) {
-                    SelectBox.add_to_cache(to, {value: option.value, text: option.text, displayed: 1});
-                    SelectBox.delete_from_cache(from, option.value);
+            $('#' + from + ' option:selected').each(function(idx, option) {
+                var value = option.value;
+
+                if (SelectBox.cache_contains(from, value)) {
+                    SelectBox.add_to_cache(to, {value: value, text: option.text, displayed: 1});
+                    SelectBox.delete_from_cache(from, value);
                 }
-            }
+            });
             SelectBox.redisplay(from);
             SelectBox.redisplay(to);
         },
         move_all: function(from, to) {
-            var from_box = document.getElementById(from);
-            var option;
-            var boxOptions = from_box.options;
-            for (var i = 0, j = boxOptions.length; i < j; i++) {
-                option = boxOptions[i];
-                if (SelectBox.cache_contains(from, option.value)) {
-                    SelectBox.add_to_cache(to, {value: option.value, text: option.text, displayed: 1});
-                    SelectBox.delete_from_cache(from, option.value);
+            $('#' + from + ' option').each(function(idx, option) {
+                var value = option.value;
+
+                if (SelectBox.cache_contains(from, value)) {
+                    SelectBox.add_to_cache(to, {value: value, text: option.text, displayed: 1});
+                    SelectBox.delete_from_cache(from, value);
                 }
-            }
+            });
```

Comment:
The thing is there's no such thing as a `:selected` selector, it's a [`jQuery`](https://api.jquery.com/selected-selector/) extension. You end up iterating over you collection of `option`s in both cases.